### PR TITLE
feat: Change RateModelClient => ConfigStoreClient to handle new AcrossConfigStore

### PR DESCRIPTION
**Pull request by nicholaspai**
RateModelClient and ConfigStoreClient are folded into the same `AcrossConfigStoreClient` now. Previously hardcoded config vars like `MAX_RELAYER_REPAYMENT_LEAF_SIZE` and `MAX_POOL_REBALANCE_LEAF_SIZE` are now queried from on-chain `AcrossConfigStore` contract.

### CODE REVIEW

File: src/clients/RateModelClient.ts [hunk 1]
```diff
@@ -65,33 +78,100 @@ export class RateModelClient {
...
+        }
+
+        // Store RateModel:
+        // TODO: Temporarily reformat the shape of the event that we pass into the sdk.rateModel class to make it fit
+        // the expected shape. This is a fix for now that we should eventually replace when we change the sdk.rateModel
+        // class itself to work with the generalized ConfigStore.
+        const l1Token = args.key;
+        delete args.value;
+        delete args.key;
+        this.cumulativeRateModelUpdates.push({ ...args, rateModel: rateModelForToken, l1Token });
```
Comment (chrismaree) [inline src/clients/RateModelClient.ts:145]: this is a good temp solution to get us up and running without waisting too much time. good work!

File: src/interfaces/ConfigStore.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { BigNumber } from "../utils";
+import { SortableEvent } from "./Common";
+export interface L1TokenTransferThreshold extends SortableEvent {
```
Comment (chrismaree) [inline src/interfaces/ConfigStore.ts:3]: excellent.

File: src/interfaces/ConfigStore.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { BigNumber } from "../utils";
+import { SortableEvent } from "./Common";
```
Comment (chrismaree) [inline src/interfaces/ConfigStore.ts:2]: it might make sence to join `Common` with `index.ts` for these interfaces but we can change this later.

**Review by chrismaree — APPROVED** [review]
this is great! your implementation means that we dont need to make any SDK or protocol changes (right now) and can refactor. this once everything else is done. good work!

File: src/clients/RateModelClient.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/RateModelClient.ts:145]: yeah eventually will change the SDK

File: src/clients/RelayerClientHelper.ts [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
 import { getProvider, getSigner, getDeployedContract, getDeploymentBlockNumber, Contract } from "../utils";
-import { SpokePoolClient, HubPoolClient, RateModelClient, TokenClient, ProfitClient, MultiCallerClient } from ".";
+import {
+  SpokePoolClient,
+  HubPoolClient,
+  AcrossConfigStoreClient,
+  TokenClient,
+  ProfitClient,
+  MultiCallerClient,
+} from ".";
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/clients/RelayerClientHelper.ts:10]: will fix!

File: src/interfaces/ConfigStore.ts (hunk 3 above)
Comment (nicholaspai) [reply src/interfaces/ConfigStore.ts:2]: How does that work? You mean just add them to `index.ts`?

**Pull request merged**